packages/trpc-interface/src/shared/client.ts

```diff
@@ -9,6 +9,7 @@ import { callerLink } from "../trpc-caller-link";
 type SharedClientOptions = {
   url: string;
   token: string;
+  headers: Record<string, string>;
```

Comment:
instead of passing headers, I would explicitly pass the commit and build the header inside createTrpcProxyServiceClient, so that it has a more narrow api